pandas/core/apply.py

```diff
@@ -1267,6 +1270,9 @@ def series_generator(self) -> Generator[Series, None, None]:
                 ser._mgr = mgr
                 mgr.set_values(arr)
                 object.__setattr__(ser, "_name", name)
```

Comment:
To be honest, I don't understand how this part adds an extra ref after the first row (i.e. the reason you need to reset `refs` to have one ref to only itself)

Reply:
>  I didn’t want to screw with that

It's already taking a shallow copy always (even when not needed), so having a specialized shallow copy without tracking refs should at best slightly improve things. 

The logic about requiring this shallow copy also already lives where the series_generator is used, so I think I find it slightly clearer to consolidate that logic there

Now, actually implementing a `Series._shallow_copy_without_refs()` is not fully trivial because AFAIK we can't just reuse the existing mgr and block `copy` methods and reset the refs after the fact, as that then already appended the refs object which is shared with the series generator series with a new ref. 
So let's consider that for a follow-up PR if I want to pursue that ;)

Reply:
Yeah I'd rather not do too many helpers that make this easy since you really shouldn't be doing this